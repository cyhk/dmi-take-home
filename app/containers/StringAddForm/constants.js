export const ADD_STRING = 'ADD_STRING';
export const ADD_STRING_SUCCESS = 'ADD_STRING_SUCCESS';
export const ADD_STRING_FAILURE = 'ADD_STRING_FAILURE';
