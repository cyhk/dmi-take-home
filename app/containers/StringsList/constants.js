export const GET_STRINGS = 'GET_STRINGS';
export const GET_STRINGS_SUCCESS = 'GET_STRINGS_SUCCESS';
export const GET_STRINGS_FAILURE = 'GET_STRINGS_FAILURE';
