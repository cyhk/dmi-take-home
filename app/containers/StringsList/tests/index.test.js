/**
 * Test the StringsList
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { StringsList, mapDispatchToProps } from '../index';
import { GET_STRINGS } from '../constants';
import configureStore from '../../../configureStore';

describe('<StringsList />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringsList
            strings={null}
            fetching={false}
            error={null}
            getStrings={jest.fn()}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('getStrings', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.getStrings).toBeDefined();
      });

      it('should dispatch GET_STRINGS when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.getStrings();
        expect(dispatch).toHaveBeenCalledWith({ type: GET_STRINGS });
      });
    });
  });
});
