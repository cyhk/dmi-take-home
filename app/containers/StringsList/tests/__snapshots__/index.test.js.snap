// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringsList /> should render and match the snapshot 1`] = `null`;
